logagg_collector/collector_command.py

```diff
@@ -0,0 +1,108 @@
+import os
+import socket
+
+from basescript import BaseScript
+from deeputil import AttrDict
+import tornado.ioloop
+import tornado.web
+from kwikapi.tornado import RequestHandler
+from kwikapi import API
+from logagg_utils import start_daemon_thread
+
+from .collector import LogCollector, CollectorService
+
+class LogaggCollectorCommand(BaseScript):
+    DESC = 'Logagg command line tool'
+
+    def parse_master_args(self, master_arg):
+        '''
+        Parse master arguments
+        '''
+        # Collector running with or without master
+        if not self.args.no_master:
+            master = AttrDict()
+            try:
+                m = self.args.master.split(':')
+                # So that order of keys is not a factor
+                for a in m:
+                    a = a.split('=')
+                    if a[0] == 'host': master.host = a[-1]
+                    elif a[0] == 'port': master.port = a[-1]
+                    elif a[0] == 'topic_name': master.topic_name = a[-1]
+                    else: raise ValueError
+
+            except ValueError:
+                raise Exception('Invalid argument arg: {}'.format(self.args.master))
+            return master
+
+        else:
+            return None
+
+    def collect(self):
+        '''
+        Start collector service after parsing arguments
+        '''
+        master = self.parse_master_args(self.args.master)
+        # Create collector object
+        collector = LogCollector(self.args.host,
+                                    self.args.port,
+                                    master,
+                                    self.args.data_dir,
+                                    self.args.logaggfs_dir,
+                                    self.log)
+        # Request authentication master details 
+        register_response = collector.register_to_master()
```

Comment:
register master is raising some errors in some cases but we are not catching them here please do check this.